test/chainrules.jl

```diff
@@ -3,6 +3,14 @@
         x = [randn(1, 2), randn(2, 2)]
         test_rrule(BlockDiagonal, x)
```

Comment:
What's the automatically derived tangent in this case? Doesn't look like we have `ProjectTo` defined, so is it the structural?

Reply:
It's a BlockDiagonal (we do have ProjectTo defined at the bottom of `chainrules.jl` file).

also, testing thunks is now automatic feature of ChainRulesTestUtils

Reply:
Lovely. That makes like easy then.